R/class-Node.R

```diff
@@ -0,0 +1,88 @@
+check_node <- function(object) {
+
+  errors <- character()
+
+  # Node color must be a string or number
+  if (!is.null(object@color) & !is.character(object@color) & !is.numeric(object@color)) {
+    errors <- c(errors, "Node color must be a string or number")
+  }
+
+  # Node weight must be a number
+  if (!is.null(object@weight) & !is.numeric(object@weight)) {
+    errors <- c(errors, "Node weight must be a number")
+  }
+
+  return(if (length(errors) == 0) TRUE else errors)
+}
+
+
+#' Node
+#' 
+#' A class for representing nodes in a network
+#' 
+#' @slot id string a unique identifier for the node
+#' @slot color string or numeric that determines the color of the node. Optional.
+#' @slot weight numeric value associated with the node, such as timestamp or other node-associated data. Optional.
+#' 
+#' @name Node-class
+#' @rdname Node-class
+#' @export
+Node <- setClass("Node", 
+  representation(
+    id = "character",
+    color = "ANY",
```

Comment:
I'm not sure how we're planning to manage colors yet, it's unclear how you mean this to be used.

Reply:
I'm wondering if this isn't an inherent property of the network but maybe instead a method param? Maybe the network can just hold any relevant annotations/ data about the nodes and the method can worry about color scheme etc

Reply:
i think we're on the same page and that's where i'm going with this idea. The `linkColorScheme` slot is just there to let someone who gets the network know how the colors were mapped. Like if i said here map these values to colors you'd want to know should it be an ordinal, categorical, diverging, etc. color map to use later. This slot says "hey btw i got colored according to this principle, so use that info to render me appropriately.